### PR TITLE
Fix petsc spec file so verification test case works with Spectrum MPI

### PR DESCRIPTION
**Pull request by dwootton-ny**
Running 'spack install --test=root petsc' using Spectrum MPI as the MPI provider fails running the ex50 testcase in the verification step.

The verification step issues 'mpirun ex50 -da_grid_x 4 -da_grid_y 4' which fails with the message '[0]PETSC ERROR: Argument out of range
[0]PETSC ERROR: Partition in y direction is too fine! 4 11' on my system, where the mpirun default is to run with 44 tasks since I have 44 cores available.

By adding '-np 4' to the mpirun command, a reasonable number of tasks is specified and ex50 runs.

### CODE REVIEW

**Comment by dwootton-ny** [comment]
Updated comments in petsc package.py and undid core reformatting

**Comment by dwootton-ny** [comment]
The build failed after my last commit due to line length > 79. I hesitated about lines > 80 characters but then saw that previous lines exceeded that length. Also, the only URL I can find that describes Spectrum MPI
behavior ends up at 112 characters with proper indentation.

I can easily fix the line flake8 complains about, but what do I do about the URL?

**Comment by adamjstewart** [comment]
@dwootton-ny Spack ignores certain lines when testing line length. This includes URLs. See https://github.com/spack/spack/blob/develop/lib/spack/spack/cmd/flake8.py#L59,L103 for a list of all exemptions.

Rest assured that if you fix the line it is complaining about and `spack flake8` passes locally, you can safely ignore all other line lengths.

File: var/spack/repos/builtin/packages/petsc/package.py [hunk 1]
```diff
@@ -296,6 +296,17 @@ def install(self, spec, prefix):
                 cc('ex50.c', '-I%s' % prefix.include, '-L%s' % prefix.lib,
                    '-lpetsc', '-lm', '-o', 'ex50')
                 run = Executable(join_path(spec['mpi'].prefix.bin, 'mpirun'))
+                # For Spectrum MPI, if -np is omitted, the default behavior is
+                # to assign one process per process slot, where the default
+                # process slot allocation is one per core. On systems with
+                # many cores, the number of processes can exceed the size of
+                # the grid specified when the testcase is run and the test case
+                # fails. Specify a small number of processes to prevent failure.
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/petsc/package.py]: This is the only line that is too long.

**Comment by dwootton-ny** [comment]
The comment line length problem should be fixed.

**Comment by dwootton-ny** [comment]
Is there anything else I need to do so this fix can be merged?

**Comment by dwootton-ny** [comment]
@citibeth I want to get this change merged so it doesn't get lost. Is there anything else I need to do to get this change merged? Thanks.

**Comment by dwootton-ny** [comment]
@adamjstewart Now that the changes are approved, can you merge this pull request? Thanks

**Pull request merged**